lambda_function.py

```
@@ -3,223 +3,290 @@
+
+def clamp_index(trajectories, idx: int) -> int:
+    """Clamp the manual reference index to the available trajectory range."""
+    if not trajectories:
+        raise ValueError("No trajectory data available.")
+    return max(0, min(len(trajectories) - 1, idx))
+
+
+def calculate_affine_params(manual_refs: dict, trajectories: list):
+    """Solve the 2D affine transform that aligns Metashape X/Z to world EN coordinates."""
     if len(manual_refs) < 3:
         raise ValueError("At least 3 reference points are required for an affine transformation.")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)